style(aio): updated padding-right to h3 in _heading-anchors.scss

haoRchen opened the pull request:
The h3 element is overflowing over its surrounding div element. Modified padding-left to align consistently with the remainder of div contents.

fixes: #22407

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
<img width="409" alt="issue" src="https://user-images.githubusercontent.com/19397242/36632838-c33255ee-1959-11e8-85bf-175a74f1b656.png">

Issue Number: 22407


## What is the new behavior?
<img width="410" alt="fixed" src="https://user-images.githubusercontent.com/19397242/36632863-f1e24944-1959-11e8-9a60-15b10ea2b4e1.png">


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

haoRchen committed:
style(aio): added padding-left to h3 in _subsection.scss

The h3 element is overflowing over its surrounding div element. Modified padding-left to align consistently with the remainder of div contents.

fixes: #22407

haoRchen commented:
In addition, I noticed that the header is not being aligned properly in mobile/responsive views. See below:

<img width="438" alt="mobile view error" src="https://user-images.githubusercontent.com/19397242/36632213-bcee11b2-1951-11e8-8fbc-c1b1fc9d0987.png">

Please advise.

gkalpak commented on aio/src/styles/2-modules/_subsection.scss:
```diff
@@ -11,6 +11,7 @@
 
     h3 {
         margin: 8px 0 0;
+        padding-left: 34px;
```
As mentioned by @petebacondarwin in https://github.com/angular/angular/issues/22407#issuecomment-368243696, the styles for `<h3>` are fine.

You should instead change the `.heading-anchor` styles in [_heading-anchors.scss][1]. More specifically, I think you should change [this line][2]:

```diff
padding-right: 24px;
padding-right: 64px;
```

(For reference, this was accidentally broken in e3e7044d0.)

[1]: https://github.com/haoRchen/angular/blob/5ee251ea0f1fde91b163eb2df08abf8b9d761236/aio/src/styles/2-modules/_heading-anchors.scss
[2]: https://github.com/haoRchen/angular/blob/5ee251ea0f1fde91b163eb2df08abf8b9d761236/aio/src/styles/2-modules/_heading-anchors.scss#L34

haoRchen commented:
I've applied your fix @gkalpak, please let me know if theres anything else I can do. See below for some updated screenshots: 
<img width="411" alt="fix" src="https://user-images.githubusercontent.com/19397242/36651030-a812f90a-1a74-11e8-918b-23d2decb0619.png">
<img width="435" alt="side fix" src="https://user-images.githubusercontent.com/19397242/36651036-ab9814b6-1a74-11e8-848e-6a1022f97ea0.png">

mary-poppins commented:
You can preview 5ee251e at https://pr22431-5ee251e.ngbuilds.io/.

mary-poppins commented:
You can preview 8a429a6 at https://pr22431-8a429a6.ngbuilds.io/.

SeanPrashad commented:
After looking further into other documentation with @haoRchen, we stumbled across another section in his [patch mentioned above](https://pr22431-8a429a6.ngbuilds.io/) that has a formatting issue [here](https://pr22431-8a429a6.ngbuilds.io/guide/dependency-injection-in-action#take-a-break):

<img width="1218" alt="post-dev-suggestion-fix" src="https://user-images.githubusercontent.com/13009507/36651398-cb6b84ec-1a76-11e8-87c6-fb52c712c1a1.png">

haoRchen committed:
style(aio): updated padding-right for the .alert class in _heading-anchors.scss

The h3 element is overflowing over its surrounding div element. Modified padding-right to align consistently with the remainder of div contents.

fixes: #22407

haoRchen commented:
https://github.com/angular/angular/pull/22431/commits/2dbbeaad3b4617ffcb6e6e3c18132a63a54d7c95 addresses [@SeanPrashad 's comment](https://github.com/angular/angular/pull/22431#issuecomment-368376667). 

After fix screenshot:
<img width="522" alt="fix for take a break" src="https://user-images.githubusercontent.com/19397242/36652134-b0f15b56-1a7a-11e8-9630-5094e2047ff2.png">

mary-poppins commented:
You can preview 2dbbeaa at https://pr22431-2dbbeaa.ngbuilds.io/.

mary-poppins commented:
You can preview cde3dc1 at https://pr22431-cde3dc1.ngbuilds.io/.

mary-poppins commented:
You can preview 6647bae at https://pr22431-6647bae.ngbuilds.io/.
